pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/onchistory/OncHistoryUploadService.java

```diff
@@ -125,23 +129,38 @@ public void upload(String fileContent) {
     /**
      * Given participant short IDs in uploaded rows, verify the short ID, and get and record associated
      * participant IDs and medical record IDs
+     * verify if any exited participant(s) exists in passed onc history records.
      *
-     * @throws OncHistoryValidationException for failed verification
+     * @throws OncHistoryValidationException for failed verifications
      */
     protected Map<Integer, Integer> getParticipantIds(List<OncHistoryRecord> oncHistoryRecords,
                                                       ParticipantIdProvider participantIdProvider, boolean updateElastic) {
         Map<Integer, Integer> medIds = new HashMap<>();
+        List<String> exitedParticipants = new ArrayList<>();
 
         ParticipantDao participantDao = ParticipantDao.of();
 
+        String esIndex = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow().getEsParticipantIndex();
         for (OncHistoryRecord rec : oncHistoryRecords) {
-            int participantId = participantIdProvider.getParticipantIdForShortId(rec.getParticipantTextId());
+            ElasticSearchParticipantDto ptpData = elasticSearchService.getParticipantDocumentByShortId(
+                    rec.getParticipantTextId(), esIndex).orElseThrow(() -> new OncHistoryValidationException("Invalid short ID " + rec.getParticipantTextId()));
+            if (ptpData.getStatus().isPresent() && ptpData.getStatus().get().startsWith("EXITED")) {
+                exitedParticipants.add(rec.getParticipantTextId());
+            }
+
+            if (!exitedParticipants.isEmpty()) {
+                //skip MR verify/creation. continuing to collect any other exited participant hruids.
+                continue;
+            }
 
+            int participantId;
             try {
+                participantId = ptpData.getDsm().get().getParticipant().get().getParticipantId().intValue();
```

Comment:
Combining all this in one line removes error detail provided in `ESParticipantIdProvider`.
Perhaps break this out into a separate method that replicates the code in `ESParticipantIdProvider.getParticipantIdForShortId`.